app/policies/business_objective_policy.rb

```diff
@@ -3,20 +3,17 @@
 # Policies for accessing BusinessObjectives
 class BusinessObjectivePolicy < ApplicationPolicy
   def index?
-    record.profiles.pluck(:account_id).include? user.account_id
+    record.in_account(user.account_id)
   end
 
   def show?
-    record.profiles.pluck(:account_id).include? user.account_id
+    record.in_account(user.account_id)
   end
 
   # Only show hosts in our user account
   class Scope < ::ApplicationPolicy::Scope
     def resolve
-      ids = scope.all.select do |business_objective|
-        business_objective.profiles.pluck(:account_id).include? user.account_id
-      end
-      scope.where(id: ids)
```

Comment:
This part was flawed in the same way as above, but it also has the added problems of using ruby-based select and a `where` with an unbounded number of ids, which might also be problematic for maximum SQL query length.

Reply:
This is the real issue I'd say - the query is nearly identical for the `index?` and `show?` before and after, it gets all account IDs through `profiles`, then checks if the current account_id is there. Thanks @akofink